substrate/frame/asset-rewards/src/benchmarking.rs

```diff
@@ -0,0 +1,442 @@
+		);
+
+		assert_ok!(AssetRewards::<T>::create_pool(
+			Root.into(),
+			Box::new(staked_asset.clone()),
+			Box::new(reward_asset.clone()),
+			100u32.into(),
+			200u32.into(),
+			None,
+		));
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(staker.clone()), 0u32.into(), 100u32.into());
+
+		assert_last_event::<T>(
+			Event::Staked { who: staker, pool_id: 0u32.into(), amount: 100u32.into() }.into(),
+		);
+	}
+
+	#[benchmark]
+	fn unstake() {
+		use super::*;
+
+		let staker = T::BenchmarkHelper::to_account_id([1u8; 32]);
+		let staked_asset = T::BenchmarkHelper::to_asset_id(1);
+		let reward_asset = T::BenchmarkHelper::to_asset_id(0);
+		create_and_mint_asset::<T>(
+			&staker,
+			&staked_asset,
+			T::Assets::minimum_balance(staked_asset.clone()),
+		);
+		create_and_mint_asset::<T>(
+			&staker,
+			&reward_asset,
+			T::Assets::minimum_balance(reward_asset.clone()),
+		);
+
+		assert_ok!(AssetRewards::<T>::create_pool(
+			Root.into(),
+			Box::new(staked_asset.clone()),
+			Box::new(reward_asset.clone()),
+			100u32.into(),
+			200u32.into(),
+			None,
+		));
+
+		assert_ok!(AssetRewards::<T>::stake(
+			RawOrigin::Signed(staker.clone()).into(),
+			0u32.into(),
+			100u32.into()
+		));
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(staker.clone()), 0u32.into(), 100u32.into());
+
+		assert_last_event::<T>(
+			Event::Unstaked { who: staker, pool_id: 0u32.into(), amount: 100u32.into() }.into(),
+		);
+	}
+
+	#[benchmark]
+	fn harvest_rewards() {
+		use super::*;
+
+		let block_number_before = frame_system::Pallet::<T>::block_number();
+		let staker = T::BenchmarkHelper::to_account_id([2u8; 32]);
+		let staked_asset = T::BenchmarkHelper::to_asset_id(1);
+		let reward_asset = T::BenchmarkHelper::to_asset_id(2);
+		create_and_mint_asset::<T>(
+			&staker,
+			&staked_asset,
+			T::Assets::minimum_balance(staked_asset.clone()),
+		);
+		create_and_mint_asset::<T>(
+			&staker,
+			&reward_asset,
+			T::Assets::minimum_balance(reward_asset.clone()),
+		);
+		assert_ok!(AssetRewards::<T>::create_pool(
+			Root.into(),
+			Box::new(staked_asset.clone()),
+			Box::new(reward_asset.clone()),
+			100u32.into(),
+			200u32.into(),
+			None,
+		));
+
+		let pool_acc = AssetRewards::<T>::pool_account_id(&0u32).unwrap();
+		create_and_mint_sufficient::<T>(&pool_acc);
+		create_and_mint_asset::<T>(&pool_acc, &reward_asset, 100_000u32.into());
+		assert_ok!(AssetRewards::<T>::stake(
+			RawOrigin::Signed(staker.clone()).into(),
+			0u32.into(),
+			100u32.into()
+		));
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(staker.clone()), 0u32.into(), None);
+
+		let block_number_after = frame_system::Pallet::<T>::block_number();
+
+		// In tests a block doesn't pass but when running benchmarks for node-template one does.
+		// Not sure why, but adding this to correctly calculate the harvested amount.
+		//
+		// TODO: Before merging understand this
+		let blocks_elapsed = block_number_after - block_number_before;
```

Comment:
@ggwpez this is where `blocks_elapsed` in the test differs from `kitchensink`. 

Let me know if we should create an issue.